src/processors/AsyncAPIInputProcessor.ts

```diff
@@ -30,8 +30,8 @@ export class AsyncAPIInputProcessor extends AbstractInputProcessor {
     }
     inputModel.originalInput = doc;
     // Go over all the message payloads and convert them to models
-    for (const [, message] of doc.allMessages()) {
-      const schema = AsyncAPIInputProcessor.convertToInternalSchema(message.payload());
+    for (const [messageName, message] of doc.allMessages()) {
+      const schema = AsyncAPIInputProcessor.convertToInternalSchema(message.payload(), messageName);
```

Comment:
So this change in this file, I kinda understand, but not really, do you mind clarifying why this was introduced? 🤔 

The reason why I am confused about this change is that when calling `schema.uid();` the message name should already have been applied to the payload schema by the parser 🤔

Reply:
This change is why many of the snapshots now all of a sudden expect `LessAnonymousMinusMessageMinus_1Greater` instead of `AnonymousMessage_1`.

Reply:
This was the only change I added that's not related to `oneOf`. And also why I stated that I wanted to do other changes in separate MRs. I can remove it, but I added it because it made the name of the union `PetMessage` instead of `Anonymous`. For some reason, the parser doesn't add the name to the message.

Reply:
Alright, let me take a look at the parser and figure out where we need to adapt this behavior then. Cause I know we have quite a few issues when it comes to the naming of the data models.

I am 100% with you it should not be `Anonymous`, but I don't think this is how we fix it 🙂